### PR TITLE
Fix the bug when deleting a page.

### PR DESCRIPTION
**Pull request by emmguyot**
Fix the bug when deleting a page. 
This bug is the same as reported in #14 but it occurs with release 4.2

### CODE REVIEW

**Comment by emmguyot** [comment]
Sorry, the problem is wider : If the first access is from BE after the clean cache operation, then the code isn't beautified in FE.
I need to investigate a bit more and update this pull request

**Comment by fnagel** [comment]
Thanks for the PR! Even if its not the final solution ;-)

Your second issue looks a lot like #17. Take a look, you might find some useful information.

**Comment by emmguyot** [comment]
You're right. I've forgotten this issue I investigated 2 years ago.
I'll comment there until I've finished my patch

**Comment by emmguyot** [comment]
I think it's ok : It solves both #14 and #17

File: Classes/Domain/Model/Flexform.php [hunk 1]
```diff
@@ -81,6 +81,28 @@ class Flexform extends \TYPO3\CMS\Extbase\DomainObject\AbstractValueObject
...
+     *
+     * @param array $flexformValues parsed values
+     * sample :
+     * cLabel:""
+     * cLang:"cpp"
+     * cHighlight:""
+     * cCollapse:"auto"
+     * cGutter:"auto"
+     */
+    public function initializeObject(array $flexformValues = array())
```
Comment (fnagel) [inline Classes/Domain/Model/Flexform.php:95]: You need to use the new array syntax `$flexformValues = []`

**Review by fnagel — CHANGES_REQUESTED** [review]

**Comment by fnagel** [comment]
@emmguyot Did you find a way to test this bug reliable? I've seen the error on my own website but was never able to reproduce it in order to investigate. Any hint how to test this?

**Comment by emmguyot** [comment]
> @emmguyot Did you find a way to test this bug reliable? I've seen the error on my own website but was never able to reproduce it in order to investigate. Any hint how to test this?

Sure, I'm using it on my website and it works fine.
What kind of error have you got ?

My config : PHP 7.4.15, Typo3 10.4.15

**Comment by emmguyot** [comment]
If you want to reproduce the #14 and #17 bugs, here are what I've done.

#14 : with previous release
- Make sure that the Highlighter works fine from the FE
- It means that in BE / System / Configuration / $GLOBALS['TCA'] you'll find tx_beautyofcode_domain_model_flexform
- Create a page
- Delete it and the error occurs

#17 : With a release where there is a condition in tx_beautyofcode_domain_model_flexform.php whether we are in BE ou not (As I've tried at first in this PR)
- Flush caches
- Access or create a content for this plugin
- Typo3 will now consider his cache as ok and won't parse again for tx_beautyofcode_domain_model_flexform from the php file
- Access from the FE a page where a content with this plugin appears
- You get the source code in 'plain' format because no language is set (As I explained in a comment of the bug)

**Comment by fnagel** [comment]
> What kind of error have you got ?

Error as in no JS powered highlighting in place, at least from time to time. Sounds like the bug described in #17 

> My config : PHP 7.4.15, Typo3 10.4.15

My config looks quite similar.


Regarding the bugs:

> #14 : with previous release

Same thing, but I was confident this was fixed back in the days. I remember reproducing and fixing it. But it might occur again, but my TYPO3 10 instance works just fine with current master.


> #17 : ...

Help me out here. When you say "Access or create a content for this plugin" you mean create a new instance of the plugin? Or modifying an existing one? I was not able to reproduce this, neither with or without BE auth.

**Review by fnagel — APPROVED** [review]
+1 by reading

I will need to test this locally, but code wise and from reading it looks good now!

**Comment by emmguyot** [comment]
> > #14 : with previous release
> 
> Same thing, but I was confident this was fixed back in the days. I remember reproducing and fixing it. But it might occur again, but my TYPO3 10 instance works just fine with current master.

Well I had the problem with the 4.2.0 release. According to the commits since that release, there is no way it may have changed with the current master. So it looks strange.
I haven't tested it with a clean fresh install of Typo3, maybe that's the point.

> > #17 : ...
> 
> Help me out here. When you say "Access or create a content for this plugin" you mean create a new instance of the plugin? Or modifying an existing one? I was not able to reproduce this, neither with or without BE auth.

Access > Means only open the BE to an existing instance of the plugin. It's enough to make Typo3 build the cache of $GLOBALS['TCA']
Create > Enter the form to create a new instance of the plugin.

If you're using the current master, you won't get the bug. But to fix the #14 as it was previously fixed, it reveals this bugs again. Check previous commits where test was done in ext_tables.php (e.g. line 9 of commit d3550bb5643581588fa8da871de66ddb6c40c90c) to build the $GLOBALS['TCA'] according to FE ou BE access. At first, I've tried to set back this test that disappeared from commit f5da486addbc2f10959893c1633a816c487b63ee. And then, I got this bug again.

**Comment by fnagel** [comment]
Sorry, I'm a little confused here. Commit f5da486addbc2f10959893c1633a816c487b63ee has never been merged actually. And I'm really not sure which version of the code I need to checkout. Diff between master and the 4.2.0 release shows no relevant changes for those issues.

Let me summarize this. We have two bugs:

1) The "code is not highlighted" bug #17 
2) The "page delete" bug #14

My questions are:
Which bug is is still present in current master and / or latest release (which is 4.2.0)?
Which bug does this PR in its current state fix?

**Comment by emmguyot** [comment]
The bug #14 is present with the latest release.

**Comment by fnagel** [comment]
So issues #17 is not addressed in this PR? I will try to reproduce #14 again in the 4.2.0 release.

**Comment by emmguyot** [comment]
Right, the issue #17 isn't present in the master so this PR doesn't address it.
But this PR changes how #17 was fixed, that's why I mentioned it. 
Anyway, feel free to change the description.

**Comment by fnagel** [comment]
Ahh okay, I think I misunderstood because #17 is not fixed yet.

Anyway, I was able to reproduce #14 in the latest 4.2.0 release by the following steps: create page, create plugin, remove page. I'm using TYPO3 10.4.21 with PHP 7.4.12.

Any idea if this already occurred in TYPO3 9.x?

**Comment by fnagel** [comment]
Merged manually (minor changes and commit message adjustment) in fed4dd2182107d9c58872cfcc847854fed486ea0

Thanks for the contribution!

**Comment by emmguyot** [comment]
> Any idea if this already occurred in TYPO3 9.x?

I don't know

**Pull request closed without merging**